### PR TITLE
feat: Add builds section to publisher pages

### PR DESCRIPTION
**Pull request by steverydz**
## Done
Moved the builds publisher section into the publishers React application

## How to QA
- You may need to register a new OAuth application (see [Hacking.md](https://github.com/canonical/snapcraft.io/blob/main/HACKING.md#github))
- Go to http://localhost:8004/<snap_name>/builds
- Check that you can connect a repo
- Check that you can disconnect a repo
- Check that you can trigger builds

## Testing
- [ ] This PR has tests
- [x] No testing required (explain why): Will do separately as this PR is already quite big

## Issue / Card
Fixes #

### CODE REVIEW

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    refetch();
+    setAutoTriggerBuild(false);
+  }
+
+  if (autoTriggerBuild) {
+    trigger();
+  }
+
+  useEffect(() => {
+    setTimeout(() => {
```
Comment (ilayda-cp) [inline static/js/publisher-pages/pages/Builds/RepoConnected.tsx:117]: Just have a question. It's purpose is to refetch the data after 30 secs when the component is rendered right?  Should it be an interval?

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    const response = await fetch(`/api/${snapId}/builds/trigger-build`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-CSRFToken": window.CSRF_TOKEN,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error("Unable to trigger new build");
```
Comment (ilayda-cp) [inline static/js/publisher-pages/pages/Builds/RepoConnected.tsx:105]: How do we handle the thrown errors? Do we have an ErrorBoundary wrapped the components?

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+                sortKey: "result",
+                className: "p-table__cell--icon-placeholder",
+              },
+              {
+                content: "Build finished",
+                sortKey: "status",
+                className: "u-align--right",
+              },
+            ]}
+            rows={data.snap_builds.map(
```
Comment (ilayda-cp) [inline static/js/publisher-pages/pages/Builds/RepoConnected.tsx:205]: should we assign it to a variable to increase the readability?

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+        </>
+      )}
+      {disconnectModalOpen && (
+        <Modal
+          close={() => {
+            setDisconnectModalOpen(false);
+          }}
+          title="Disconnecting your repository"
+          buttonRow={
+            <>
```
Comment (ilayda-cp) [inline static/js/publisher-pages/pages/Builds/RepoConnected.tsx:250]: Should we create a separate component for button row?

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+                }}
+              >
+                Cancel
+              </Button>
+              <Button
+                appearance="positive"
+                className="u-no-margin--bottom u-no-margin--right"
+                disabled={disconnecting}
+                onClick={async () => {
+                  setDisconnecting(true);
```
Comment (ilayda-cp) [inline static/js/publisher-pages/pages/Builds/RepoConnected.tsx:264]: could you create a function for this?

**Review by ilayda-cp — COMMENTED** [review]
Great work! Just left some tiny comments.

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx (hunk 5 above)
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoConnected.tsx]: Done

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx (hunk 4 above)
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoConnected.tsx]: Done

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx (hunk 3 above)
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoConnected.tsx:205]: `data` is only guaranteed to be defined within this scope so we'd need to add some logic here to handle that if we used a variable, which I don't see adds any benefits?

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx (hunk 2 above)
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoConnected.tsx:105]: We don't have a standard way of handling errors across our projects. This is something we should discuss further as a team I think.

File: static/js/publisher-pages/pages/Builds/RepoConnected.tsx (hunk 1 above)
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoConnected.tsx]: You are correct. Fixed

File: static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
...board-1/new/master?filename=snap%2Fsnapcraft.yaml&value=%0A%20%20%23%20After%20registering%20a%20name%20on%20build.snapcraft.io%2C%20commit%20an%20uncommented%20line%3A%0A%20%20%23%20name%3A%20steve-test-snap%0A%20%20version%3A%20%270.1%27%20%23%20just%20for%20humans%2C%20typically%20%271.2%2Bgit%27%20or%20%271.3.2%27%0A%20%20summary%3A%20Single-line%20elevator%20pitch%20for%20your%20amazing%20snap%20%23%2079%20char%20long%20summary%0A%20%20description%3A%20%7C%0A%20%20%20%20This%20is%20my-snap%27s%20description.%20You%20have%20a%20paragraph%20or%20two%20to%20tell%20the%0A%20%20%20%20most%20important%20story%20about%20your%20snap.%20Keep%20it%20under%20100%20words%20though%2C%0A%20%20%20%20we%20live%20in%20tweetspace%20and%20your%20description%20wants%20to%20look%20good%20in%20the%20snap%0A%20%20%20%20store.%0A%0A%20%20grade%3A%20devel%20%23%20must%20be%20%27stable%27%20to%20release%20into%20candidate%2Fstable%20channels%0A%20%20confinement%3A%20devmode%20%23%20use%20%27strict%27%20once%20you%20have%20the%20right%20plugs%20and%20slots%0A%0A%20%20parts%3A%0A%20%20%20%20my-part%3A%0A%20%20%20%20%20%20%23%20See%20%27snapcraft%20plugins%27%0A%20%20%20%20%20%20plugin%3A%20nil%0A%20%20">
```
[the remark this answers is not in the document]
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx]: Done

File: static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+                  <option value={repo.name} key={repo.name} />
+                ))}
+              </datalist>
+            </Col>
+            <Col size={2}>
+              {validRepo === true && (
+                <Button
+                  appearance="positive"
+                  disabled={building}
+                  onClick={async () => {
```
[the remark this answers is not in the document]
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx]: Done

File: static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+                options={[
+                  { label: "Select organization", value: "" },
+                  {
+                    label: githubData.github_user.name,
+                    value: githubData?.github_user.login,
+                  },
+                  ...getOrgs(),
+                ]}
+                onChange={async (e) => {
+                  if (e.target.value) {
```
[the remark this answers is not in the document]
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx]: Done

File: static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+              </li>
+            </ul>
+          </div>
+        </div>
+      )}
+      <div className="u-fixed-width">
+        <hr className="u-no-margin--bottom" />
+      </div>
+      {githubData !== null && (
+        <Strip shallow>
```
[the remark this answers is not in the document]
Comment (steverydz) [reply static/js/publisher-pages/pages/Builds/RepoNotConnected.tsx]: Done

**Review by ilayda-cp — APPROVED** [review]

**Pull request merged**